src/lib/work_partitioner/pending_zkapp_command.ml

```diff
@@ -26,7 +26,7 @@ let create ~job ~unscheduled_segments ~pending_mergeable_proofs ~merge_remaining
   ; merge_remaining
   }
 
-let generate_merge ~(t : t) () =
+let generate_merge (t : t) =
```

Comment:
Let's rename to `next_merge`?

Subjective, I know :D.

Also, could we add a doc-comment explaining that the function dequeues from `pending_mergeable_proofs` and combines two proof into a merge spec?

I want it to be clear that the function changes the state.